$MirOS: ports/misc/screen/patches/patch-etc_etcscreenrc,v 1.4 2006/10/31 03:40:13 tg Exp $
--- etc/etcscreenrc.orig	Fri Dec  5 13:45:51 2003
+++ etc/etcscreenrc	Wed Nov  8 15:53:53 2006
@@ -1,20 +1,21 @@
+# $Id$
+# $MirOS: ports/misc/screen/patches/patch-etc_etcscreenrc,v 1.4 2006/10/31 03:40:13 tg Exp $
 #
 # This is an example for the global screenrc file.
-# You may want to install this file as /usr/local/etc/screenrc.
-# Check config.h for the exact location.
+# You may want to install this file as @@SYSCONFDIR@@/screenrc.
 #
 # Flaws of termcap and standard settings are done here.
-#
 
-#startup_message off
+# Don't annoy me with FSF propaganda on startup
+startup_message off
 
 #defflow on # will force screen to process ^S/^Q
 
-deflogin on
+deflogin off
 #autodetach off
 
 vbell on
-vbell_msg "   Wuff  ----  Wuff!!  "
+vbell_msg "   Suddenly, your kitten starts purring!!  "
 
 # all termcap entries are now duplicated as terminfo entries.
 # only difference should be the slightly modified syntax, and check for
@@ -36,9 +37,13 @@ vbell_msg "   Wuff  ----  Wuff!!  "
 # IC    insert multiple chars
 # nx    terminal uses xon/xoff
 
-termcap  facit|vt100|xterm LP:G0
-terminfo facit|vt100|xterm LP:G0
+# You wouldn't believe the docs to lie, would you?
+termcap  ansi-nt am@:hs@
+terminfo ansi-nt am@:hs@
 
+termcap  facit|vt100|xterm|xterm-xfree86 LP:G0
+terminfo facit|vt100|xterm|xterm-xfree86 LP:G0
+
 #the vt100 description does not mention "dl". *sigh*
 termcap  vt100 dl=5\E[M
 terminfo vt100 dl=5\E[M
@@ -55,20 +60,20 @@ terminfo sun 'up=^K:AL=\E[%p1%dL:DL=\E[%
 #xterm understands both im/ic and doesn't have a status line.
 #Note: Do not specify im and ic in the real termcap/info file as
 #some programs (e.g. vi) will (no,no, may (jw)) not work anymore.
-termcap  xterm|fptwist hs@:cs=\E[%i%d;%dr:im=\E[4h:ei=\E[4l
-terminfo xterm|fptwist hs@:cs=\E[%i%p1%d;%p2%dr:im=\E[4h:ei=\E[4l
+termcap  xterm|xterm-xfree86|fptwist hs@:cs=\E[%i%d;%dr:im=\E[4h:ei=\E[4l
+terminfo xterm|xterm-xfree86|fptwist hs@:cs=\E[%i%p1%d;%p2%dr:im=\E[4h:ei=\E[4l
 
 # Long time I had this in my private screenrc file. But many people
 # seem to want it (jw):
 # we do not want the width to change to 80 characters on startup:
 # on suns, /etc/termcap has :is=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;3;4;6l:
-termcap xterm 'is=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;4;6l'
-terminfo xterm 'is=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;4;6l'
+termcap  xterm|xterm-xfree86 'is=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;4;6l'
+terminfo xterm|xterm-xfree86 'is=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;4;6l'
 
 #
 # Do not use xterms alternate window buffer. 
 # This one would not add lines to the scrollback buffer.
-#termcap xterm|xterms|xs ti=\E7\E[?47l
+#termcap  xterm|xterms|xs ti=\E7\E[?47l
 #terminfo xterm|xterms|xs ti=\E7\E[?47l
 
 #make hp700 termcap/info better
@@ -78,17 +83,35 @@ terminfo hp700 'Z0=\E[?3h:Z1=\E[?3l:hs:t
 #wyse-75-42 must have defflow control (xo = "terminal uses xon/xoff")
 #(nowadays: nx = padding doesn't work, have to use xon/off)
 #essential to have it here, as this is a slow terminal.
-termcap wy75-42 nx:xo:Z0=\E[?3h\E[31h:Z1=\E[?3l\E[31h
+termcap  wy75-42 nx:xo:Z0=\E[?3h\E[31h:Z1=\E[?3l\E[31h
 terminfo wy75-42 nx:xo:Z0=\E[?3h\E[31h:Z1=\E[?3l\E[31h
 
-#remove some stupid / dangerous key bindings
+# Set this to off if your system doesn't grok TERM=screen-bce
+defbce on
+
+# remove some stupid / dangerous key bindings
 bind ^k
 #bind L
 bind ^\
-#make them better
+# make them better
 bind \\ quit
 bind K kill
 bind I login on
 bind O login off
 bind } history
 
+shell "-${SHELL}"
+shelltitle -
+hardstatus alwayslastline "%{kr} %H %{ky} load[%l] %{kg} %d.%m.%Y %c %{+b}%{yb} %=%?%-w%?[%n%f %t]%?%+w%? "
+nethack on
+defscrollback 4096
+
+# set the "inner" codeset to UTF-8 by default
+defutf8 on
+setenv LC_CTYPE en_US.UTF-8
+# put this into your ~/.screenrc to revert it
+#defutf8 off
+#setenv LC_CTYPE C
+
+# set the escape character = ^P instead of ^A
+#escape ^Pp
